openlibrary/plugins/openlibrary/code.py

```diff
@@ -814,6 +816,28 @@ def GET(self):
         h = guppy.hpy()
         return delegate.RawText(str(h.heap()))
 
+def _get_relatedcarousels_component(workid):
+    work = web.ctx.site.get('/works/%s' % workid) or {}
+    component = render_template('books/RelatedWorksCarousel', work)
+    return {0: str(component)}
+
+def get_cached_relatedcarousels_component(*args, **kwargs):
+    memoized_get_component_metadata = cache.memcache_memoize(
+        _get_relatedcarousels_component, "book.bookspage.component.relatedcarousels", timeout=dateutil.HALF_DAY_SECS)
+    return (memoized_get_component_metadata(*args, **kwargs) or
+            memoized_get_component_metadata.update(*args, **kwargs)[0])
+
+class Partials(delegate.page):
+    path = '/partials'
+
+    def GET(self):
+        i = web.input(workid=None, _component=None)
+        component = i.pop("_component")
+        cached_component = {}
+        if component == "RelatedWorkCarousel":
```

Comment:
The next person who works on this should make this dispatch generic such that we don't have to manually check the value of component.

We should instead do something like `getattr(self, component)(**i)` which would automatically call the right function

```
In [2]: class Foo(object):
   ...:     def __init__(self):
   ...:         pass
   ...:     def bar(self, a=2):
   ...:         print(a)
   ...:     def baz(self):
   ...:         getattr(self, 'bar')(a=5)
   ...: 
In [3]: x = Foo()
In [4]: x.baz()
5
```